juniper/src/macros/object.rs

```diff
@@ -137,11 +137,11 @@ automatically via the `?` operator, or you can construct them yourself using
 struct User { id: String }
 
 graphql_object!(User: () |&self| {
-    field id() -> FieldResult<&String> {
+    field id() -> FieldResult<&String, __S> {
```

Comment:
I really don't like this public API. Is it possible to default `FieldResult` to the proper type in the simple case and introduce something like  `CustomScalarFieldResult` for the custom case?

Reply:
Similar to the execute function it is possible to define type aliases with default types, so the definition of `FieldResult` would look like this: `type FieldResult<T, S = DefaultScalarValue> = Result<T, FieldError<S>>;`

(Sidenote: In this case this would not have been helpful, because the `graphql_object!` macros introduces internally a generic scalar type (`__S`). This macro needs to be rewritten in a similar manner like the `graphql_scalar!` macro)

Reply:
It should be much more clearer now. 
It is either `FieldResult<&String>` for using the `DefaultScalarValue` or `FieldResult<&String, ScalarType>` for using  a custom type or a generic parameter. In both cases this type is introduced in the header of the macro.